src/sources/index.test.ts

```diff
@@ -0,0 +1,47 @@
+import { getComponents } from './index'
+
+describe('Sources', () => {
+  describe('getComponents', () => {
+    it('handles errors', async () => {
+      const sources = {
+        success1: () => 'foo',
+        throwsErrorObject: () => { throw new Error('bar') },
+        throwsErrorString: () => { throw 'baz' },
+        success2: () => 'baq',
+      }
+      const components = await getComponents(sources, undefined, [])
+      expect(Object.keys(components).sort()).toEqual(['success1', 'success2', 'throwsErrorObject', 'throwsErrorString'])
+      expect(components.success1.error).toBeUndefined()
```

Comment:
Maybe create assertion helper function, like `expectSucessfulValue(components.success1)`. And vice versa for an error result below

Reply:
I think that checking a component status is a very short and clear procedure. Adding a helper function will add a level of abstraction that may increase the cognitive complexity of the code.